velox/exec/Window.cpp

```diff
@@ -282,6 +302,128 @@ void Window::callResetPartition(vector_size_t partitionNumber) {
   }
 }
 
+void Window::updateKRowsFrameBounds(
+    bool isKPreceding,
+    const FrameChannelArg& frameArg,
+    vector_size_t startRow,
+    vector_size_t numRows,
+    vector_size_t* rawFrameBounds) {
+  auto firstPartitionRow = partitionStartRows_[currentPartition_];
+
+  if (frameArg.index == kConstantChannel) {
+    auto constantOffset = frameArg.constant.value();
+    std::iota(
+        rawFrameBounds,
+        rawFrameBounds + numRows,
+        startRow + (isKPreceding ? -constantOffset : constantOffset) -
```

Comment:
nit: I wonder if it would be easier to read if isKPreceding was processed outside of this call.

```
auto offset = frameArg.constant.value();
if (isKPreceding) {
   offset = -constantOffset;
}
std::iota(
        rawFrameBounds,
        rawFrameBounds + numRows,
        startRow - firstPartitionRow + offset);
```